Build fails due to invalid gzip option -k

### What did you do?

Attempt to build Prometheus on a CentOS 7 platform

### What did you expect to see?

A successful build. It last worked on version 2.30.3 that I've tested.

### What did you see instead? Under which circumstances?

```
gzip: invalid option -- 'k'
Try `gzip --help' for more information.
>> building binaries
/home/kitchen/recipes/prometheus/tmp-build/bin/promu build --prefix /home/kitchen/recipes/prometheus/tmp-build/src/github.com/prometheus/prometheus
 >   prometheus
# github.com/prometheus/prometheus/web/ui
web/ui/embed.go:21:3: usage: //go:embed pattern...
!! command failed: build -o /home/kitchen/recipes/prometheus/tmp-build/src/github.com/prometheus/prometheus/prometheus -ldflags -X github.com/prometheus/common/version.Version=2.37.0 -X github.com/prometheus/common/version.Revision=b41e0750abf5cc18d8233161560731de05199330 -X github.com/prometheus/common/version.Branch=HEAD -X github.com/prometheus/common/version.BuildUser=kitchen@3460a3706333 -X github.com/prometheus/common/version.BuildDate=20220721-14:47:40  -extldflags '-static' -a -tags netgo,builtinassets github.com/prometheus/prometheus/cmd/prometheus: exit status 2
make: *** [common-build] Error 1
```

This comes from:
```
scripts/compress_assets.sh:find static -type f -exec gzip -fk '{}' \; -print0 | xargs -0 -I % echo %.gz | xargs echo //go:embed >> embed.go
```

It seems when building on CentOS 7, the available gzip version does not include the `-k` option, which causes the build to fail. Builds should not be host dependent, outside of the Go / Node versions.

### System information

CentOS 7

### Prometheus version

```text
2.37.0
```


### Prometheus configuration file

```yaml
N/A
```


### Alertmanager version

```text
N/A
```


### Alertmanager configuration file

```yaml
N/A
```


### Logs

```text
N/A
```

--- a/scripts/compress_assets.sh
+++ b/scripts/compress_assets.sh
@@ -8,7 +8,9 @@ cd web/ui
 cp embed.go.tmpl embed.go
 
 # gzip option '-k' may not always exist in the latest gzip available on different distros.
-GZIP_OPTS="-fk" && [[ ! $(gzip -k -h 2>/dev/null) ]] && GZIP_OPTS="-f"
+GZIP_OPTS="-fk"
+# gzip option '-k' may not always exist in the latest gzip available on different distros.
+if ! gzip -k -h &>/dev/null; then GZIP_OPTS="-f"; fi
 
 find static -type f -name '*.gz' -delete
 find static -type f -exec gzip $GZIP_OPTS '{}' \; -print0 | xargs -0 -I % echo %.gz | xargs echo //go:embed >> embed.go

--- a/scripts/compress_assets.sh
+++ b/scripts/compress_assets.sh
@@ -6,6 +6,10 @@ set -euo pipefail
 
 cd web/ui
 cp embed.go.tmpl embed.go
+
+# gzip option '-k' does not exist on RHEL7 based instances, so remove it
+GZIP_OPTS="-fk" && [[ $(awk -F= '/^NAME/{print $2}' /etc/os-release) =~ (CentOS|Red Hat).* ]] && [[ $(awk -F= '/^VERSION_ID/{print $2}' /etc/os-release) =~ 7.* ]] && GZIP_OPTS="-f"
+
 find static -type f -name '*.gz' -delete
-find static -type f -exec gzip -fk '{}' \; -print0 | xargs -0 -I % echo %.gz | xargs echo //go:embed >> embed.go
+find static -type f -exec gzip $GZIP_OPTS '{}' \; -print0 | xargs -0 -I % echo %.gz | xargs echo //go:embed >> embed.go
 echo var EmbedFS embed.FS >> embed.go

--- a/scripts/compress_assets.sh
+++ b/scripts/compress_assets.sh
@@ -7,8 +7,8 @@ set -euo pipefail
 cd web/ui
 cp embed.go.tmpl embed.go
 
-# gzip option '-k' does not exist on RHEL7 based instances, so remove it
-GZIP_OPTS="-fk" && [[ $(awk -F= '/^NAME/{print $2}' /etc/os-release) =~ (CentOS|Red Hat).* ]] && [[ $(awk -F= '/^VERSION_ID/{print $2}' /etc/os-release) =~ 7.* ]] && GZIP_OPTS="-f"
+# gzip option '-k' may not always exist in the latest gzip available on different distros
+GZIP_OPTS="-fk" && [[ ! $(gzip -k -h 2>/dev/null) ]] && GZIP_OPTS="-f"
 
 find static -type f -name '*.gz' -delete
 find static -type f -exec gzip $GZIP_OPTS '{}' \; -print0 | xargs -0 -I % echo %.gz | xargs echo //go:embed >> embed.go

--- a/scripts/compress_assets.sh
+++ b/scripts/compress_assets.sh
@@ -7,7 +7,6 @@ set -euo pipefail
 cd web/ui
 cp embed.go.tmpl embed.go
 
-# gzip option '-k' may not always exist in the latest gzip available on different distros.
 GZIP_OPTS="-fk"
 # gzip option '-k' may not always exist in the latest gzip available on different distros.
 if ! gzip -k -h &>/dev/null; then GZIP_OPTS="-f"; fi

--- a/scripts/compress_assets.sh
+++ b/scripts/compress_assets.sh
@@ -7,7 +7,7 @@ set -euo pipefail
 cd web/ui
 cp embed.go.tmpl embed.go
 
-# gzip option '-k' may not always exist in the latest gzip available on different distros
+# gzip option '-k' may not always exist in the latest gzip available on different distros.
 GZIP_OPTS="-fk" && [[ ! $(gzip -k -h 2>/dev/null) ]] && GZIP_OPTS="-f"
 
 find static -type f -name '*.gz' -delete

